opengovsg/askgovsg
# feat(seo): sensible meta descriptions for agency and post pages

## Problem

Meta descriptions of dynamic pages currently do not reflect the content of the pages. An example is shown below:

<img width="581" alt="Screenshot 2021-09-30 at 2 50 25 PM" src="https://user-images.githubusercontent.com/37061143/135401906-133b2242-dead-4fa9-ae95-e40d037d6919.png">

Closes #394 and #22

## Solution

Dynamically populate meta descriptions for agency-specific pages to include the agency name and post pages with the answer to the question.

**Features**:

- update meta description for post page with text of first answer
- update meta description for agency-specific page with 'Answers from `agency.longname` (`agency.shortname`)' (e.g. 'Answers from Work Allocation Singapore (WAS)')
- set meta prop `data-rh` in `index.html` to `true`
- refactor AnswerSection component to prevent repeat answer queries

**Improvements**:

- refactor PageTitle component to typescript
- replace `react-helmet` with `react-helmet-async`

## Before & After Screenshots

**BEFORE**:

On agency-specific pages:

<img width="893" alt="Screenshot 2021-09-30 at 3 18 02 PM" src="https://user-images.githubusercontent.com/37061143/135405565-b1d6ca7a-cf01-4ecb-af85-568b2763f2b6.png">

On post pages:

<img width="895" alt="Screenshot 2021-09-30 at 3 17 48 PM" src="https://user-images.githubusercontent.com/37061143/135405581-5662b668-cde5-417a-967a-07014e0a9ea6.png">

**AFTER**:

On agency-specific pages:

<img width="894" alt="Screenshot 2021-09-30 at 3 13 21 PM" src="https://user-images.githubusercontent.com/37061143/135405078-f1222c83-afe0-4607-96ef-6fd60d3a2cc3.png">

On post pages:

<img width="897" alt="Screenshot 2021-09-30 at 3 13 51 PM" src="https://user-images.githubusercontent.com/37061143/135405085-28bf0f2b-c22e-4597-b5b0-e285e8b672a2.png">

## Tests

1. Navigate to an agency-specific page and check the meta description with inspect - it should be 'Answers from `agency.longname` (`agency.shortname`)'.
2. Navigate to a post page and check the meta description with inpect - it should be the first answer to the question.
3. Navigate to the main AskGov page and check the meta description with inspect - it should be 'Answers from the Singapore Government'.

## Deploy Notes

**New dependencies**:

- `dependency`: 
  - `client/package.json`: `react-helmet-async`
  - `client/package.json`: `sanitize-html`
  - `client/package.json`: `@types/sanitize-html`

**Removed dependencies**:

- `dependency`: 
  - `client/package.json`: `react-helmet`
